Implement Trello integration

angelsanzn opened the pull request:
Includes two notification strategies:

```
- card-wise: for each notification, creates a new item
in a specified checklist of a specified card.

- list-wise: for each notification, creates a new card
in a specified list.
```

Commit:
Implement Trello integration.

Includes two notification strategies:
    - card-wise: for each notification, creates a new item
    in a specified checklist of a specified card.

    - list-wise: for each notification, creates a new card
    in a specified list.

ergl commented:
Nice! Taking a look at it now

ergl commented on spec/trello_integration_spec.py:
```diff
@@ -0,0 +1,254 @@
+from doublex_expects import have_been_called_with
+
+from pysellus.stock_integrations import trello
+from pysellus.interfaces import AbstractIntegration
```
seems like `AbstractIntegration` is not used here?

angelsanzn replied:
Thanks, fixed.

ergl commented:
`CardWiseNotifier` and `ListWiseNotifier` look almost identical, do you think you could abstract them together?

Also, I see that there's only `on_next` in here — perhaps an `on_error` implementation could be useful as well?

ergl commented:
Also, for this integration, how would an config entry look like?

angelsanzn commented:
The config entry would just declare constructor parameters:

``` yaml
    'my-trello-alias':
        trello:
            key: ...
            token: ...
            card: ...
            checklist: ...
            mode: list
            list: ...
```

As long as you declare the required parameters (and only those) for the selected (or default) mode, it works.

ergl commented:
Just getting back to this now — as I said, everything looks good, but I think that the two strategies can be abstracted a bit, right now they look too similar.

What do you think?

angelsanzn commented:
Yes, I refactored in order to tighten the responsibilities of each class and pushed between yesterday and just a few moments ago. What do you think of it as it stands now? I believe it's ready.

(I also implemented `on_error` and `on_completed`.)

ergl commented:
Yeah, this looks much better, good job!

angelsanzn commented:
Thanks!
